admin/config/locales/en.yml

```diff
@@ -286,6 +286,7 @@ en:
             views: views
         edit:
           client_id_note: Identifies the category for product import
+          default_sort_note: 'When a category has featured products, this sort will appear as the "Featured" option on the storefront.'
```

Comment:
If we only output this when there are featured products, this can be really specific and helpful like: "Since there are featured products, this will determine the sort of products showing after the featured products." Or something explicit like that.